phoenix-ios/phoenix-ios/AppDelegate.swift

```diff
@@ -339,6 +341,52 @@ class AppDelegate: UIResponder, UIApplicationDelegate, MessagingDelegate {
 		}
 	}
 
+	// --------------------------------------------------
+	// MARK: Background Execution
+	// --------------------------------------------------
+	
+	func setupActivePaymentsListener() -> Void {
+		
+		business.paymentsManager.inFlightOutgoingPaymentsPublisher().sink { [weak self](count: Int) in
+			
+			log.debug("inFlightOutgoingPaymentsPublisher: count = \(count)")
+			if count > 0 {
```

Comment:
Should not we also check that `isInBackground` is true ? Otherwise this background task would be started every time the `PaymentManager.inFlightOutgoingPayments` flow changes ?

Reply:
Wow, Apple made this super confusing. I did a double-take myself when re-reading the code...

The original "background task" was more like a long-lived task (that should continue running even if the app is put into the background). It was designed for situations where "leaving a task unfinished may cause a bad user experience in your app."

(_For example, if you're making a messaging app, and the user sends a message, then they expect the message to properly get sent, even if they immediately switch to another app._)

And then, in iOS 13, Apple introduced a new BackgroundTask framework, designed for different kinds of background tasks. And thus the confusion...

So I renamed the variables & functions in this PR to "longLivedTask". Which is less confusing to me.